platform/android/src/ch/opengis/qfield/QFieldGalleryPictureActivity.java

```diff
@@ -66,13 +66,24 @@ protected void onActivityResult(int requestCode, int resultCode,
 
         if (requestCode == GALLERY_ACTIVITY) {
             if (resultCode == RESULT_OK) {
-                File result = new File(prefix + pictureFilePath);
+                Uri uri = data.getData();
+                DocumentFile documentFile = DocumentFile.fromSingleUri(
+                    getApplication().getApplicationContext(), uri);
+
+                String pictureName = documentFile.getName();
+                String pictureExtension = "";
+                if (pictureName.indexOf('0') > -1) {
+                    pictureExtension =
+                        pictureName.substring(pictureName.indexOf('0') + 1);
+                }
 
+                String finalPictureFilePath = pictureFilePath;
+                finalPictureFilePath.replace("#~filename~#", pictureName);
+                finalPictureFilePath.replace("#~extension~#", pictureExtension);
```

Comment:
Could we put that code into a function used by both Activities (I think it would help to dedup and to increase readability because right now I'm not completely sure what it does)